frontend/src/metabase/ui/components/inputs/MultiAutocomplete/MultiAutocomplete.tsx

```diff
@@ -53,40 +59,84 @@ export function MultiAutocomplete({
+
+    const values = parseValues(searchValue);
+    const validValues = values.filter(isValid);
+
     setSearchValue("");
+
+    if (values.length > 0) {
+      const newValues = [...lastSelectedValues, ...validValues];
+      setSelectedValues(newValues);
+      setLastSelectedValues(newValues);
+    } else {
+      setLastSelectedValues(selectedValues);
+    }
+
     onBlur?.(event);
   };
 
-  const handleSearchChange = (newSearchValue: string) => {
-    setSearchValue(newSearchValue);
+  const handlePaste = (event: ClipboardEvent<HTMLInputElement>) => {
+    event.preventDefault();
+
+    const input = event.target as HTMLInputElement;
+    const value = input.value;
+    const before = value.slice(0, input.selectionStart ?? value.length);
+    const after = value.slice(input.selectionEnd ?? value.length);
+
+    const pasted = event.clipboardData.getData("text");
+    const text = `${before}${pasted}${after}`;
 
-    const isValid = shouldCreate?.(newSearchValue, selectedValues);
-    if (isValid) {
-      setSelectedValues([...lastSelectedValues, newSearchValue]);
+    const values = parseValues(text);
+    const validValues = values.filter(isValid);
+
+    if (values.length > 0) {
+      const newValues = [...lastSelectedValues, ...validValues];
+      setSelectedValues(newValues);
+      setLastSelectedValues(newValues);
+      setSearchValue("");
     } else {
-      setSelectedValues(lastSelectedValues);
+      setSearchValue(text);
+      setSelectedValues([...lastSelectedValues, text]);
     }
   };
 
-  const handlePaste = (event: ClipboardEvent<HTMLInputElement>) => {
-    const text = event.clipboardData.getData("Text");
-    const values = text.split(/[\n,]/g);
-    if (values.length > 1) {
-      const validValues = [...new Set(values)]
-        .map(value => value.trim())
-        .filter(value => shouldCreate?.(value, selectedValues));
-      if (validValues.length > 0) {
-        event.preventDefault();
-        const newSelectedValues = [...lastSelectedValues, ...validValues];
-        setSelectedValues(newSelectedValues);
-        setLastSelectedValues(newSelectedValues);
+  const handleSearchChange = (newSearchValue: string) => {
+    const last = newSearchValue.at(-1);
+
+    setSearchValue(newSearchValue);
+    setSelectedValues([...lastSelectedValues, newSearchValue]);
+
+    if (last === "," || last === "\t" || last === "\n" || last === '"') {
+      const values = parseValues(newSearchValue);
+      const validValues = values.filter(
+        value => value !== "" && shouldCreate?.(value, lastSelectedValues),
+      );
+
+      if (values.length > 0) {
+        const newValues = [...lastSelectedValues, ...validValues];
+        setSelectedValues(newValues);
+        setLastSelectedValues(newValues);
+
+        setSearchValue("");
       }
     }
   };
 
+  const info = (
+    <Tooltip
+      label={t`Separate values with commas, tabs or newlines. Use double quotes for values containing commas.`}
```

Comment:
nit: The tooltip is very wide. IMO it'd look a bit better in 2 lines. Maybe centered too.
WDYT?

![image](https://github.com/metabase/metabase/assets/6830683/34331538-d5ed-4b1f-ba03-5f7b365da72b)